bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/api/EnedisHttpApi.java

```diff
@@ -0,0 +1,225 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.linky.internal.api;
+
+import java.net.CookieStore;
+import java.net.HttpCookie;
+import java.net.URI;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.util.FormContentProvider;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.util.Fields;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.openhab.binding.linky.internal.LinkyConfiguration;
+import org.openhab.binding.linky.internal.LinkyException;
+import org.openhab.binding.linky.internal.dto.AuthData;
+import org.openhab.binding.linky.internal.dto.AuthResult;
+import org.openhab.binding.linky.internal.dto.ConsumptionReport;
+import org.openhab.binding.linky.internal.dto.ConsumptionReport.Consumption;
+import org.openhab.binding.linky.internal.dto.PrmInfo;
+import org.openhab.binding.linky.internal.dto.UserInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * {@link EnedisHttpApi} wraps the Enedis Webservice.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class EnedisHttpApi {
+    private static final DateTimeFormatter API_DATE_FORMAT = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+    private static final String URL_APPS_LINCS = "https://apps.lincs.enedis.fr";
+    private static final String URL_MON_COMPTE = "https://mon-compte.enedis.fr";
+    private static final String URL_ENEDIS_AUTHENTICATE = URL_APPS_LINCS
+            + "/authenticate?target=https://mon-compte-particulier.enedis.fr/suivi-de-mesure/";
+    private static final String URL_COOKIE = "https://mon-compte-particulier.enedis.fr";
+
+    private final Logger logger = LoggerFactory.getLogger(EnedisHttpApi.class);
+    private final Gson gson;
+    private final HttpClient httpClient;
+
+    public EnedisHttpApi(LinkyConfiguration config, Gson gson, HttpClient httpClient) throws LinkyException {
+        this.gson = gson;
+        this.httpClient = httpClient;
+
+        httpClient.getSslContextFactory().setExcludeCipherSuites(new String[0]);
+        httpClient.setFollowRedirects(false);
+
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            throw new LinkyException("Unable to start Jetty HttpClient", e);
+        }
+        connect(config);
```

Comment:
Consider moving this logic to a separate `initialize` method for this class. It is good practice to separate construction/configuration logic from initialization logic.

Reply:
Done